lib/miq_expression.rb

```diff
@@ -1563,37 +1563,41 @@ def self._quick_search?(e)
     end
   end
 
-  def custom_attribute_columns_and_models(expression = nil)
-    return custom_attribute_columns_and_models(exp).uniq if expression.nil?
+  def custom_attribute_columns_and_models
+    @virtual_custom_attributes ||= find_custom_attributes_and_models_recursively || []
```

Comment:
WDYT about this being the reader for this attribute? i.e. `@custom_attribute_columns_and_models`?